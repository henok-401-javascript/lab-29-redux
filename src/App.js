import React from 'react';
import Header from './components/Header';
import Footer from './components/Footer';
import {Provider} from 'react-redux';
import store from './Store';
import Products from './components/Products';
import Category from './components/Categories';
 import Container from '@material-ui/core/Container';
import Card from '@material-ui/core/Card';

function App() {
  return (
    <div className="App">
      <Container maxWidth="sm">
      <Card>
      <Provider store={store}>
  <Header/>
  <Category/>
  <Products/>
  <Footer/>
      </Provider>
      </Card>
      </Container>
    </div>
  );
}

export default App;
