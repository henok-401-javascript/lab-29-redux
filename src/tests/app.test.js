import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer'

describe('Testing by using snapshot' , () =>{
  it('It successfully check the UI' , () =>{
    let currentHTML = renderer.create(<App/>).toJSON();
    expect(currentHTML).toMatchSnapshot();
  })
})