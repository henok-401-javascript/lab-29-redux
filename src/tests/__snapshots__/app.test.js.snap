// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing by using snapshot It successfully check the UI 1`] = `
<div
  className="App"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <header
        className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        OUR STORE
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            CART(0)
          </span>
        </button>
      </header>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Electronics
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Food
          </span>
        </button>
      </div>
      <div>
        <h4>
          Apple
           
        </h4>
        <h4>
          Pizza
           
        </h4>
        <h4>
          Kitfo
           
        </h4>
      </div>
      <div>
        Â© our store.
      </div>
    </div>
  </div>
</div>
`;
